statsmodels/formula/_manager.py

```diff
@@ -0,0 +1,723 @@
+
+        Parameters
+        ----------
+        constraints
+        variable_names
+
+        Returns
+        -------
+        LinearConstraintValues
+            The constraint matrix, constraint values, and variable names.
+        """
+        if self._engine == "patsy":
+            from patsy.design_info import DesignInfo
+
+            lc = DesignInfo(variable_names).linear_constraint(constraints)
+            return LinearConstraintValues(
+                constraint_matrix=lc.coefs,
+                constraint_values=lc.constants,
+                variable_names=lc.variable_names,
+            )
+        else:  # self._engine == "formulaic"
+
+            # Handle list of constraints, which is not supported by formulaic
+            if isinstance(constraints, list):
+                if len(constraints) == 0:
+                    raise ValueError("Constraints must be non-empty")
+
+                if isinstance(constraints[0], str):
+                    if not all(isinstance(c, str) for c in constraints):
+                        raise ValueError(
+                            "All constraints must be strings when passed as a list."
+                        )
+                    _constraints = ", ".join(str(v) for v in constraints)
+                else:
+                    _constraints = np.array(constraints)
+            else:
+                _constraints = constraints
+            if isinstance(_constraints, tuple):
+                _constraints = (
+                    _constraints[0],
+                    np.atleast_1d(np.squeeze(_constraints[1])),
+                )
+            lc_f = formulaic.utils.constraints.LinearConstraints.from_spec(
+                _constraints, variable_names=list(variable_names)
+            )
+            return LinearConstraintValues(
+                constraint_matrix=lc_f.constraint_matrix,
+                constraint_values=np.atleast_2d(lc_f.constraint_values).T,
+                variable_names=lc_f.variable_names,
+            )
+
+    def get_empty_eval_env(self):
+        """
+        Get an empty evaluation environment.
+
+        Returns
+        -------
+        {EvalEnvironment, dict}
+            A formula-engine-dependent empty evaluation environment.
+        """
+        if self._engine == "patsy":
+            from patsy.eval import EvalEnvironment
+
+            return EvalEnvironment({})
+        else:
+            return {}
+
+    @property
+    def intercept_term(self):
+        """
+        Get the formula-engine-specific intercept term.
+
+        Returns
+        -------
+        Term
+            The intercept term.
+        """
+        if self._engine == "patsy":
+            from patsy.desc import INTERCEPT
+
+            return INTERCEPT
+        else:
+            from formulaic.parser.types.factor import Factor
+            from formulaic.parser.types.term import Term
+
+            return Term((Factor("1"),))
```

Comment:
Since you are only using this to check for equality, you can actually just return `"1"` here.